Signal/src/ViewControllers/SignalsViewController.m

```diff
@@ -491,6 +492,7 @@ - (void)viewDidAppear:(BOOL)animated {
         // Start running the disappearing messages job in case the newly registered user
         // enables this feature
         [[OWSDisappearingMessagesJob sharedJob] startIfNecessary];
+        [[OWSProfileManager sharedManager] ensureLocalProfileCached];
```

Comment:
I feel like this doesn't belong in a VC - it belongs in a "user did register" method in the app delegate.  Thoughts?  Not for this PR of course.

Reply:
I agree that we should extract the app setup process. maybe the appDelegate is the right spot.